src/client/src/Client.c

```diff
@@ -462,11 +590,27 @@ STATUS stopKinesisVideoStreamSync(STREAM_HANDLE streamHandle)
     CHK_STATUS(semaphoreAcquire(pKinesisVideoStream->pKinesisVideoClient->base.shutdownSemaphore, INFINITE_TIME_VALUE));
     releaseClientSemaphore = TRUE;
 
+    DLOGW("Acquired pKinesisVideoStream->pKinesisVideoClient->base.shutdownSemaphore");
+
     CHK_STATUS(semaphoreAcquire(pKinesisVideoStream->base.shutdownSemaphore, INFINITE_TIME_VALUE));
     releaseStreamSemaphore = TRUE;
 
+    DLOGW("Acquired pKinesisVideoStream->base.shutdownSemaphore");
+
+
+    // lock streams list lock
+    pKinesisVideoStream->pKinesisVideoClient->clientCallbacks.lockMutexFn(pKinesisVideoStream->pKinesisVideoClient->clientCallbacks.customData,
+                                                                            pKinesisVideoStream->pKinesisVideoClient->base.streamListLock);
+    streamsListLock = TRUE;
+
     CHK_STATUS(stopStreamSync(pKinesisVideoStream));
 
+    if(streamsListLock) {

```

Comment:
spacing and remove the if statement. If you reach here then this value will be invariant will always be TRUE